mitosheet/src/components/taskpanes/SnowflakeImport/SnowflakeImportTaskpane.tsx

```diff
@@ -0,0 +1,478 @@
+import MultiToggleItem from "../../elements/MultiToggleItem";
+import Select from "../../elements/Select";
+import TextButton from "../../elements/TextButton";
+import Col from "../../layout/Col";
+import CollapsibleSection from "../../layout/CollapsibleSection";
+import Row from "../../layout/Row";
+import Spacer from "../../layout/Spacer";
+import DefaultEmptyTaskpane from "../DefaultTaskpane/DefaultEmptyTaskpane";
+import DefaultTaskpane from "../DefaultTaskpane/DefaultTaskpane";
+import DefaultTaskpaneBody from "../DefaultTaskpane/DefaultTaskpaneBody";
+import DefaultTaskpaneHeader from "../DefaultTaskpane/DefaultTaskpaneHeader";
+
+
+interface SnowflakeImportTaskpaneProps {
+    mitoAPI: MitoAPI;
+    userProfile: UserProfile;
+    setUIState: React.Dispatch<React.SetStateAction<UIState>>;
+    analysisData: AnalysisData;
+    sheetDataArray: SheetData[];
+    selectedSheetIndex: number;
+}
+
+export type SnowflakeCredentialsValidityCheckResult = {type: 'success'} | {type: 'error', 'error_message': string}
+
+export type SnowflakeCredentials = {type: 'username/password', username: string, password: string, account: string};
+
+export type SnowflakeConnection = {
+    warehouse: string | undefined, 
+    database: string | undefined, 
+    schema: string | undefined,
+    table: string | undefined
+}
+
+export type SnowflakeConnectionAPIReturn = {
+    warehouse: string | null, 
+    database: string | null, 
+    schema: string | null,
+    table: string | null
+}
+
+export type SnowflakeQueryParams = {
+	columns: string[],
+	limit: number | undefined
+}
+
+export type SnowflakeConfigOptions = {
+	warehouses: string[],
+	databases: string[],
+	schemas: string[],
+	tables: string[]
+	columns: string[]
+}
+
+export interface SnowflakeImportParams {
+    credentials: SnowflakeCredentials,
+    connection: SnowflakeConnection,
+    query_params: SnowflakeQueryParams,
+}
+
+const getDefaultParams = (): SnowflakeImportParams | undefined => {
+    return {
+        credentials: {type: 'username/password', username: '', password: '', account: ''},
+        connection: {warehouse: undefined, database: undefined, schema: undefined, table: undefined},
+        query_params: {columns: [], limit: undefined},
+    }
+}
+
+export type AvailableSnowflakeOptionsAndDefaults = {
+    type: 'success',
+    config_options: SnowflakeConfigOptions,
+    default_connection_values: SnowflakeConnectionAPIReturn
+} | {
+    type: 'error',
+    error_message: string
+}
+
+
+/* 
+    This is the SnowflakeImport taskpane.
+*/
+const SnowflakeImportTaskpane = (props: SnowflakeImportTaskpaneProps): JSX.Element => {
+
+    //loading, edit, editApplied
+    const {params, setParams, edit} = useSendEditOnClick<SnowflakeImportParams, undefined>(
+            () => getDefaultParams(),
+            StepType.SnowflakeImport, 
+            props.mitoAPI,
+            props.analysisData,
+    )
+    
+    const [openCredentialsSection, setOpenCredentialsSection] = useState(true);
```

Comment:
`openCredentialsSection` sounds like it's saying _which_ credentialsSection is open (aka, that there's multiple). A more clear name would be `credentialsSectionIsOpen` I think!